### ISSUE TITLE
GCM Nonce field can be removed from rabin.vss/EncryptedDeal

### ISSUE THREAD

**Issue by coventry** [dedis/kyber#379]
[...since its value is always 0.](https://github.com/dedis/kyber/blob/89ca87/share/vss/rabin/vss.go#L224) 

```golang
	nonce := make([]byte, gcm.NonceSize())
	dealBuff, err := protobuf.Encode(d.deals[i])
	if err != nil {
		return nil, err
	}
	encrypted := gcm.Seal(nil, nonce, dealBuff, d.hkdfContext)
```

It's probably OK to leave the nonce constant because the key is ephemeral, but if so the field is just deadweight on the protocol's bandwidth.

**Comment by jeffallen** [comment]
For backwards compatibility reasons, we would not remove field Nonce from struct EncryptedDeal. It would be interesting if we could leave it set to nil, so that protobuf would not encode it, then modify ProcessEncryptedDeal to say "if e.Nonce == nil then e.Nonce = make([]byte, gcm.NonceLength())" in order to re-create the zero nonce at the receiver side.

However, this would also be a backwards incompatible change, because if a newer vss-speaker was talking to an older vss-speaker, a nil nonce could be passed into gcm.Open, resulting in a panic: https://golang.org/src/crypto/cipher/gcm.go#L194

So this is a good idea, but we won't be able to do it in v3.

**Comment by pierluca** [comment]
Fixed in #524
